src/sisl/geom/_neighbors/_finder.py

```diff
@@ -3,17 +3,213 @@
 # file, You can obtain one at https://mozilla.org/MPL/2.0/.
 from __future__ import annotations
 
-from typing import Optional, Sequence, Tuple, Union
+from functools import cached_property
+from typing import Literal, Optional, Sequence, Tuple, Union
 
 import numpy as np
 
-from sisl import Geometry
+from sisl import Geometry, Lattice
 from sisl.typing import AtomsIndex
 from sisl.utils import size_to_elements
 
 from . import _operations
 
 
+class Interactions:
+
+    def __init__(self, geometry: Geometry, finder_results, split_indices=None):
+        self.geometry = geometry
+        self.finder_results = finder_results
+        self._split_indices = split_indices
```

Comment:
should `finder_results` be hidden? Hmm...

Reply:
Yes, I think so